singleton.py

```diff
@@ -0,0 +1,21 @@
+import unittest
+
+
+class Singleton(type):
+    def __init__(cls, name, bases, attrs, **kwargs):
+        super().__init__(name, bases, attrs)
+        cls._has_instance = None
+
+    def __call__(cls, *args, **kwargs):
+        if cls._has_instance is None:
+            cls._has_instance=super().__call__(*args, **kwargs)
+        return cls._has_instance
+
+
```

Comment:
Maybe we could add a different class that uses this Singleton metaclass, do something in the init of that class and initialise a couple of instances to see if what you did in init happens or not and for which instances. I think those would be more explicit tests.